frontend/src/components/search/SaveSearchSelector.tsx

```diff
@@ -0,0 +1,156 @@
+import clsx from "clsx";
+import { usePrevious } from "src/hooks/usePrevious";
+import { useSearchParamUpdater } from "src/hooks/useSearchParamUpdater";
+import { useUser } from "src/services/auth/useUser";
+import { obtainSavedSearches } from "src/services/fetch/fetchers/clientSavedSearchFetcher";
+import { SavedSearchRecord } from "src/types/search/searchRequestTypes";
+import { searchToQueryParams } from "src/utils/search/searchFormatUtils";
+
+import { useTranslations } from "next-intl";
+import {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useState,
+} from "react";
+import { Select } from "@trussworks/react-uswds";
+
+import SimplerAlert from "src/components/SimplerAlert";
+import Spinner from "src/components/Spinner";
+
+export const SaveSearchSelector = ({
+  newSavedSearches,
+  savedSearches,
+  setSavedSearches,
+}: {
+  newSavedSearches: string[];
+  savedSearches: SavedSearchRecord[];
+  setSavedSearches: Dispatch<SetStateAction<SavedSearchRecord[]>>;
+}) => {
+  const t = useTranslations("Search.saveSearch");
+  const { user } = useUser();
+  const { searchParams, replaceQueryParams } = useSearchParamUpdater();
+  const prevSearchParams = usePrevious(searchParams);
+
+  const [selectedSavedSearch, setSelectedSavedSearch] = useState<string>();
+  const [loading, setLoading] = useState<boolean>(false);
+  const [apiError, setApiError] = useState<Error | null>();
+
+  // allows us to avoid resetting the select option when params change after selection
+  const [applyingSavedSearch, setApplyingSavedSearch] =
+    useState<boolean>(false);
+
+  const handleFetchError = useCallback((e: Error) => {
```

Comment:
My reading / thinking of `useCallback()` is for expensive calculations or functions that will be passed down as props so a child doesn't re-render. The docs say there is [no harm in using them](https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere). 

For me caching state updates and fetches feels like an over-optimization, but most of all want to be consistent about it, so if we adopt here, would like to have a follow-up to update [OpportunitySaveUserControl.tsx](https://github.com/HHS/simpler-grants-gov/blob/main/frontend/src/components/user/OpportunitySaveUserControl.tsx) to be consistent.

Reply:
thanks for bringing this up. In this case we can just get rid of the function definition altogether, since it really only needs to get used as part of the fetch function. If this was a case where the function was needed across multiple other memoized pieces of functionality the useCallback would be necessary though, otherwise you end up in a situation with infinite rerenders.